src/history.cc

```diff
@@ -494,7 +494,7 @@ commodity_history_impl_t::find_price(const commodity_t& source,
     const commodity_t * u_comm = get(namemap, u);
     const commodity_t * v_comm = get(namemap, v);
 
-#if defined(REVERSE_PREDECESSOR_MAP)
+#if REVERSE_PREDECESSOR_MAP
```

Comment:
@jwiegley can you share some context on this define, which I wasn't able to find anywhere else in the codebase?

What is its intent?
Can it be removed?

Reply:
I can't find mention of it anywhere, not even Google. So yes, I think it can be removed.

Reply:
Could it be related to [Boost's `dijkstra_shortest_paths`](https://www.boost.org/doc/libs/1_72_0/libs/graph/doc/dijkstra_shortest_paths.html)?

Reply:
Oh, that does ring a bell...

Reply:
In your mind what is a better way forward, @jwiegley?
Leave the code parts referencing `REVERSE_PREDECESSOR_MAP` as is or delete them and make a not of it, so that should you or someone else familiar remember the intent can provide context and details and it can be implemented as desired or necessary?

Reply:
It was introduced in 72fc1824d01bb4fe50405ed183afb57b0e129d69, but was apparently just an idea and never used. I'd say it's safe to delete that code.

Reply:
✅ Done 😃